pkg/util/helper/binding.go

```diff
@@ -125,20 +127,28 @@ func (a *Dispenser) TakeByWeight(w ClusterWeightInfoList) {
 }
 
 // GetStaticWeightInfoListByTargetClusters constructs a weight list by target cluster slice.
-func GetStaticWeightInfoListByTargetClusters(tcs []workv1alpha2.TargetCluster) ClusterWeightInfoList {
+func GetStaticWeightInfoListByTargetClusters(tcs, scheduled []workv1alpha2.TargetCluster) ClusterWeightInfoList {
 	weightList := make(ClusterWeightInfoList, 0, len(tcs))
-	for _, result := range tcs {
+	for _, targetCluster := range tcs {
+		var lastReplicas int32
+		for _, scheduledCluster := range scheduled {
+			if targetCluster.Name == scheduledCluster.Name {
+				lastReplicas = scheduledCluster.Replicas
+				break
+			}
+		}
 		weightList = append(weightList, ClusterWeightInfo{
-			ClusterName: result.Name,
-			Weight:      int64(result.Replicas),
+			ClusterName:  targetCluster.Name,
+			Weight:       int64(targetCluster.Replicas),
+			LastReplicas: lastReplicas,
 		})
 	}
 	return weightList
 }
 
 // SpreadReplicasByTargetClusters divides replicas by the weight of a target cluster list.
 func SpreadReplicasByTargetClusters(numReplicas int32, tcs, init []workv1alpha2.TargetCluster) []workv1alpha2.TargetCluster {
-	weightList := GetStaticWeightInfoListByTargetClusters(tcs)
+	weightList := GetStaticWeightInfoListByTargetClusters(tcs, init)
```

Comment:
The `SpreadReplicasByTargetClusters` is used by `DynamicWeight` case. 
Do you mean we have to keep the assignment stable?

Reply:
yes, from our previous implementation, DynamicWeight is just a special StaticWeight